lib/services/role.go

```diff
@@ -3288,14 +3291,56 @@ func (set RoleSet) GetAllowedSearchAsRoles() []string {
 	}
 	for _, role := range set {
 		for _, a := range role.GetSearchAsRoles(types.Allow) {
```

Comment:
iiuc the filter applies to the base role so you can do it in the outer loop
```suggestion
	for _, role := range set {
		if slices.ContainsFunc(allowFilters, func(filter SearchAsRolesOption) bool {
			return !filter(role)
		}) {
			// Don't consider this base role if it's filtered out.
			continue
		}
		for _, a := range role.GetSearchAsRoles(types.Allow) {
```